base/bucket_gocb.go

```diff
@@ -1413,10 +1413,34 @@ func (bucket CouchbaseBucketGoCB) WriteUpdateWithXattr(k string, xattrKey string
 		var writeErr error
 		// If this is a tombstone, we want to delete the document and update the xattr
 		if deleteDoc {
-			deleteBody := len(value) > 0
-			if deleteBody {
-				LogTo("CRUD+", "WriteUpdateWithXattr attempting to delete body and update the xattr for key=%s. Existing value: %s", k, value)
+			removeCas, removeErr := bucket.Remove(k, cas)
+			if removeErr == nil {
+				// Successful removal - update the cas for the xattr operation
+				cas = removeCas
+			} else if removeErr == gocb.ErrKeyNotFound {
+				// Document body has already been removed - continue to xattr processing w/ same cas
+			} else if isRecoverableGoCBError(removeErr) {
+				// Recoverable error - retry WriteUpdateWithXattr
+				continue
+			} else {
+				// Non-recoverable error - return
+				return emptyCas, removeErr
+			}
+
+			// update xattr only
+			casOut, writeErr := bucket.WriteCasWithXattr(k, xattrKey, exp, cas, nil, updatedXattrValue)
+			if writeErr != nil && writeErr != gocb.ErrKeyExists && !isRecoverableGoCBError(writeErr) {
+				LogTo("CRUD", "Update of new value during WriteUpdateWithXattr failed for key %s: %v", k, writeErr)
+				return emptyCas, writeErr
 			}
+
+			// If there was no error, we're done
+			if writeErr == nil {
```

Comment:
Might be clearer if this code block was moved above the other `if writeErr != nil && writeErr != gocb.ErrKeyExists && !isRecoverableGoCBError(writeErr) ` block